src/dbos.ts

```diff
@@ -1695,6 +1712,18 @@ export class DBOS {
 
           const wfctx = assertCurrentWorkflowContext();
 
+          // Set child deadline if a timeout is configured
+          // Explicitly set to undefined in the context (set through withWorkflowTimeout()) means "detach the child from the parent's deadline"
+          let deadlineEpochMS: number | undefined = wfctx.deadlineEpochMS;
+          if (
```

Comment:
This code should also be simpler if we are checking for null instead of explicit undefined.